File: app/frontend/ea-project/src/models/input/CreateRoleInput.js
import BaseInput from "./BaseInput";

export default class CreateRoleInput extends BaseInput {
  constructor(name = '', description = '', creationDate = new Date(), lazyLoad = false) {
    super(lazyLoad);
    this.name = name;
    this.description = description;
    this.creationDate = creationDate;
  }
}

File: app/backend/src/main/java/eaproject/input/CreateRoleInput.java
package eaproject.input;

import java.io.Serializable;
import java.sql.Timestamp;

public class CreateRoleInput extends BaseInput implements Serializable {
    private String name;
    private String description;
    private Timestamp creationDate;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Timestamp getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Timestamp creationDate) {
        this.creationDate = creationDate;
    }
}
